### PR TITLE
Add GitHub Actions based Continuous Integration

### PR DESCRIPTION
**Pull request by traversaro**
This configuration is finally working, and so I think we can start merging it. 

There are a few issues left: 
* https://github.com/robotology/idyntree/issues/569
* https://github.com/robotology/idyntree/issues/274

Some code is not useful now but will be useful as soon as those issues are fixed, so I think it is ok to keep it there.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      run: |
+        brew install ace boost eigen swig qt5 orocos-kdl
+    
+    - name: Dependencies [Ubuntu]
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        sudo apt-get install git build-essential cmake libace-dev coinor-libipopt-dev  libboost-system-dev libboost-filesystem-dev libboost-thread-dev liborocos-kdl-dev libeigen3-dev swig qtbase5-dev qtdeclarative5-dev qtmultimedia5-dev libxml2-dev liburdfdom-dev libtinyxml-dev liburdfdom-dev liboctave-dev python-dev valgrind 
+        
+    - name: Source-based Dependencies [Windows] 
+      if: matrix.os == 'windows-latest-disable-until-issue-569-is-solved'
```
Comment (diegoferigo) [inline .github/workflows/ci.yml]: Is this step triggered? ^^

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    
+    - name: Configure [Windows]
+      # Use bash also on Windows (otherwise cd, mkdir, ... do not work)
+      if: matrix.os == 'windows-latest'
+      shell: bash
+      run: |
+        mkdir -p build
+        cd build
+        # ipopt disabled until https://github.com/robotology/idyntree/issues/274 is fixed
+        cmake -A x64 -DCMAKE_TOOLCHAIN_FILE=${VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install -DIDYNTREE_COMPILE_TESTS:BOOL=ON -DIDYNTREE_USES_YARP:BOOL=OFF -DIDYNTREE_USES_ICUB_MAIN:BOOL=OFF -DIDYNTREE_RUN_VALGRIND_TESTS:BOOL=ON -DIDYNTREE_USES_PYTHON:BOOL=OFF -DIDYNTREE_USES_OCTAVE:BOOL=OFF -DIDYNTREE_USES_IPOPT:BOOL=OFF -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
```
Comment (diegoferigo) [inline .github/workflows/ci.yml]: Maybe a newline for each option would improve readability?

Comment (traversaro) [reply .github/workflows/ci.yml]: Good idea, I need to check how to do that, probably being yml is always the same strategy.

Comment (diegoferigo) [reply .github/workflows/ci.yml]: You can use the \ character and go to a newline (even indented)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (traversaro) [reply .github/workflows/ci.yml]: I found this to be more efficient then manually commenting all those `.yml` lines. : D

File: .github/workflows/ci.yml (hunk 2 above)
Comment (traversaro) [reply .github/workflows/ci.yml]: In the end I did not used a newline for each option, but I grouped a few options  for each line, but I guess that is personal taste.

Comment (diegoferigo) [reply .github/workflows/ci.yml]: Yes, looks better, thanks!

**Pull request merged**